### PR TITLE
RHBPMS-5127, RHBPMS-5128: Tasks: Empty task list loads indefinitely. …

### PR DESCRIPTION
**Pull request by nmirasch**
…Tasks: Pagination broken

### CODE REVIEW

**Comment by nmirasch** [comment]
@dgutierr here we go again.

I added some test to try the code that ensure the totalRowCountNonTrimmed has the expected value. One of the doubts here is if after the the https://github.com/dashbuilder/dashbuilder/commit/8d3fdeb03c9b9bdffb1762563fba16ef21087533 and https://github.com/dashbuilder/dashbuilder/commit/df2060a40e8cf462490181bda3fb15112f733b42 The way that we use to add group operation have to be changed. The datasetlookup added at the test reflects our usecase. Could you take a look there? Thanks a lot!

File: dashbuilder-backend/dashbuilder-dataset-sql/src/main/java/org/dashbuilder/dataprovider/sql/SQLDataSetProvider.java [hunk 1]
```diff
@@ -740,7 +740,9 @@ public DataSet run() throws Exception {
                     ResultSet _results = logSQL(_query).fetch();
                     List<DataColumn> columns = calculateColumns(groupOp);
                     DataSet dataSet = _buildDataSet(columns, _results);
-                    if (trim) dataSet.setRowCountNonTrimmed(totalRows);
+                    if (trim && postProcessingOps.isEmpty()){
```
Comment (kkufova) [inline dashbuilder-backend/dashbuilder-dataset-sql/src/main/java/org/dashbuilder/dataprovider/sql/SQLDataSetProvider.java]: Please use a correct formatting (_KIE Java Conventions_).

File: dashbuilder-backend/dashbuilder-dataset-sql/src/test/java/org/dashbuilder/dataprovider/sql/SQLDataSetTrimTest.java [hunk 2]
```diff
@@ -31,4 +42,41 @@ public void testTrim() throws Exception {
         subTest.testTrim();
         subTest.testDuplicatedColumns();
     }
+
+    @Test
+    public void testAvoidDuplicatedGroupColumn() throws Exception {
+
```
Comment (kkufova) [inline dashbuilder-backend/dashbuilder-dataset-sql/src/test/java/org/dashbuilder/dataprovider/sql/SQLDataSetTrimTest.java]: Please remove unnecessary empty lines at the beginning and at the end of the method.

**Comment by nmirasch** [comment]
@kkufova I applied the code formatting on the test class, but if I  do that on SQLDataSetProvider.java.. if results in 108 changes, In this moment, I think it would do the review more complicated, even we don't know this commit is the best solution. After the @dgutierr confirmation for that, I can apply the formatting to that class. Do you think it makes sense?

**Comment by kkufova** [comment]
@nmirasch, I agree, thank you! :+1:

**Comment by nmirasch** [comment]
@kkufova Thanks!

**Comment by dgutierr** [comment]
@kkufova Can you please confirm the dashbuilder build works fine. including the tests?

It would be good to also execute the DBs integration tests => https://github.com/dashbuilder/dashbuilder/tree/0.5.x/dashbuilder-backend/dashbuilder-dataset-sql-tests

**Comment by mareknovotny** [comment]
@dgutierr @kkufova why is not triggering the CI PR builder for this instead of manually launching integration tests?

**Comment by mareknovotny** [comment]
Jenkins test this, please

**Comment by mbiarnes** [comment]
Jenkins please retest this

**Comment by kkufova** [comment]
@dgutierr, sure, I'll check it locally.

**Comment by mareknovotny** [comment]
@kkufova please fyi 4 tests failed in resurrected Pr builder CI job

**Comment by mbiarnes** [comment]
@mareknovotny I am now with the dashbuilder 0.5.x PR - but it is not yet stable . it downloads somehow 0.9.x artifacts - and this shouldn't be

**Comment by mbiarnes** [comment]
Jenkins please retest this

**Comment by kkufova** [comment]
The build does not fail locally. I'll check the DBs integration tests tomorrow.

**Comment by mareknovotny** [comment]
@kkufova when you are ok with this, please @dgutierr merge it, thanks in advance!

**Comment by dgutierr** [comment]
I'm merging this despite the fact jenkins is complaining with a "java.lang.ClassNotFoundException: javax.servlet.ServletContextListener" in the dataset CDI module.  Local build though is working fine both for me and @kkufova. There seems to be an issue with jenkins setup.

**Comment by mareknovotny** [comment]
ok, thanks @kkufova and @dgutierr

**Pull request merged**